lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -86,7 +86,7 @@ def unmanaged?
 
     def container_names
       regular_containers = @definition["spec"]["template"]["spec"]["containers"].map { |c| c["name"] }
-      init_containers = @definition["spec"]["template"]["spec"].fetch("initContainers", {}).map { |c| c["name"] }
+      init_containers = @definition["spec"]["template"]["spec"].fetch("initContainers", []).map { |c| c["name"] }
```

Comment:
yeah, `{}.map` and `[].map` both have the same result--do nothing and return an array.

![image](https://user-images.githubusercontent.com/4789493/28536433-91739834-7075-11e7-8d8a-6f7990026e30.png)

I changed it because `initContainers` would be an array if present